packages/function-entry/src/function-entry.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import debug from 'debug';
+import { withStyles } from 'material-ui/styles';
+import Input from './input';
+
+const log = debug('pie-elements:text-entry');
+
+const FunctionEntryStyles = {
+  'white_on_black': {
+    backgroundColor: 'black'
+  },
+  'black_on_rose': {
+    backgroundColor: 'mistyrose'
+  }
+}
+
+export class FunctionEntry extends React.Component {
+  static propTypes = {
+    session: PropTypes.object,
+    model: PropTypes.object,
+    onValueChanged: PropTypes.func,
+    classes: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.session && props.session.value || ''
+    }
+  }
+
```

Comment:
need a `componentWillReceiveProps` implementation if you're managing `session.value` in `state`?

Reply:
Unless I'm missing something, the only update that can happen to it is from inside the `onChange` function, and I'm using `setState` in there to keep it up to date.

Reply:
y - in it's current context there's a `ReactDOM.render` happening so it won't need it. can ignore.